### PR TITLE
[RPS-98] Add basic search and minimal style to site

### PR DESCRIPTION
**Pull request by ngta1**
Also covers:
- RPS-107 Show summary in search results
- RPS-106 Search again on results page
- RPS-104 Each title in results is clickable link

### CODE REVIEW

File: site/src/main/webapp/WEB-INF/jsp/errorpages/ErrorPage404.jsp [hunk 1]
```diff
@@ -1,19 +1,16 @@
+
```
Comment (wojtek-oledzki) [inline site/src/main/webapp/WEB-INF/jsp/errorpages/ErrorPage404.jsp]: no empty line please before doctype definition

File: site/src/main/resources/META-INF/hst-assembly/overrides/hippo-essentials-spring.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+
+    <!--Override the default pageable factory-->
+    <bean id="org.onehippo.cms7.essentials.components.ext.PageableFactory"
+          class="org.onehippo.cms7.essentials.components.ext.DefaultPageableFactory"/>
+
+  <!-- Default no-operation do before extension -->
```
Comment (wojtek-oledzki) [inline site/src/main/resources/META-INF/hst-assembly/overrides/hippo-essentials-spring.xml]: please, can we keep indentation consistent.

**Comment by ngta1** [comment]
Formatting corrected as requested

**Comment by wojtek-oledzki** [comment]
👍

File: repository-data/webfiles/src/main/resources/site/css/demo.css [hunk 3]
```diff
@@ -0,0 +1,98 @@
+body {
```
Comment (ziemsky) [inline repository-data/webfiles/src/main/resources/site/css/demo.css]: Unless there is a specific reason to call this file `demo.css` - how about renaming it to something that doesn't suggest that a demo/test/experimental file has sneaked into production code? `cw-main.css` or `cw-styles.css` come to mind.

File: site/src/main/java/uk/nhs/digital/ps/components/SearchComponent.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+            int offset = 0;
+
+            try {
+                String offsetString = request.getParameter("offset");
+                offset = Integer.parseInt(offsetString);
+
+            } catch (NumberFormatException e) {
+                // No offset set in this request, so use default 0
+            }
```
Comment (ziemsky) [inline site/src/main/java/uk/nhs/digital/ps/components/SearchComponent.java]: How about extracting this logic into a private method and call it here as `int offset = extractOffsetFrom(request);`? 

Maybe it's just me, but, given that this whole try/catch block takes quite a few lines, it breaks the flow when reading this method.

File: repository-data/webfiles/src/main/resources/site/css/demo.css (hunk 3 above)
Comment (wojtek-oledzki) [reply repository-data/webfiles/src/main/resources/site/css/demo.css]: that was by-design. This file should be removed as soon as Avi lands in the team :)

File: site/src/main/java/uk/nhs/digital/ps/components/SearchComponent.java (hunk 4 above)
Comment (wojtek-oledzki) [reply site/src/main/java/uk/nhs/digital/ps/components/SearchComponent.java]: done in my branch for Search :)

File: acceptance-tests/src/test/resources/features/search.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    Scenario: Clickable links
+        Given I am on the site homepage
+        When I search for a publication
+        Then I want clickable result links which take me to the publication
+
+
+    Scenario: Search term retained
+        Given I am on the site homepage
+        When I search for a publication
+                Then the results page retains the original search term
```
Comment (ziemsky) [inline acceptance-tests/src/test/resources/features/search.feature]: This line appears to have been misaligned by accident.

Comment (ngta1) [reply acceptance-tests/src/test/resources/features/search.feature]: Good spot.  will correct

File: site/src/main/webapp/WEB-INF/jsp/errorpages/ErrorPage404.jsp [hunk 6]
```diff
@@ -2,18 +2,14 @@
...
 </head>
 <body>
-<h2>Welcome to Hippo</h2>
-<p>
-  It appears that you just created an empty Hippo project from the archetype. There is nothing to show on the site yet.
-  We recommend you use
-  <a href="http://<%=request.getServerName() + ':' + request.getServerPort() + "/essentials"%>" target="_blank">Hippo's setup application</a>
-  to start developing your project.
-</p>
+<h1>Page not found!!!</h1>
```
Comment (ziemsky) [inline site/src/main/webapp/WEB-INF/jsp/errorpages/ErrorPage404.jsp]: How about leaving the exclamation marks off - the users will be stressed out enough as it is, by breaking our website ;-)

File: site/src/main/java/uk/nhs/digital/ps/components/SearchComponent.java (hunk 4 above)
Comment (ngta1) [reply site/src/main/java/uk/nhs/digital/ps/components/SearchComponent.java]: Looks like this block will change going forward.  With the Pagination ticket I'm looking at, the offset is set in a different way.

File: site/src/main/webapp/WEB-INF/jsp/errorpages/ErrorPage404.jsp (hunk 6 above)
Comment (ngta1) [reply site/src/main/webapp/WEB-INF/jsp/errorpages/ErrorPage404.jsp]: Will remove...we don't want user stress :-)  The 404 page was auto added when the essential search plugin was added to the project.  We can revisit these default error pages once styling etc has been defined.

**Pull request merged**